conn/i2c/i2c.go

```diff
@@ -83,39 +85,123 @@ func (d *Dev) Write(b []byte) (int, error) {
+	}
+	t := reflect.TypeOf(b)
+	if t.Kind() != reflect.Ptr {
+		return errors.New("i2c.DevReg.ReadRegStruct() requires a pointer to a struct")
+	}
+	tt := t.Elem()
+	if tt.Kind() != reflect.Struct {
+		return errors.New("i2c.DevReg.ReadRegStruct() requires a pointer to a struct")
+	}
+	buf := make([]byte, int(tt.Size()))
+	if err := d.Tx([]byte{reg}, buf); err != nil {
+		return err
+	}
+	return binary.Read(bytes.NewReader(buf), d.Order, b)
+}
 
 // ReadRegUint8 reads a 8 bit register.
-func (d *Dev) ReadRegUint8(reg byte) (uint8, error) {
+func (d *DevReg) ReadRegUint8(reg byte) (uint8, error) {
 	var v [1]byte
-	err := d.ReadReg(reg, v[:])
+	err := d.ReadRegRaw(reg, v[:])
 	return v[0], err
 }
 
-// ReadRegUint16 reads a 16 bit register as big endian.
-func (d *Dev) ReadRegUint16(reg byte) (uint16, error) {
+// ReadRegUint16 reads a 16 bit register.
+func (d *DevReg) ReadRegUint16(reg byte) (uint16, error) {
+	if d.Order == nil {
+		return 0, errors.New("don't know if big or little endian")
+	}
 	var v [2]byte
-	err := d.ReadReg(reg, v[:])
-	return uint16(v[0])<<8 | uint16(v[1]), err
+	err := d.ReadRegRaw(reg, v[:])
+	return d.Order.Uint16(v[:]), err
+}
+
+// ReadRegUint32 reads a 32 bit register.
+func (d *DevReg) ReadRegUint32(reg byte) (uint32, error) {
+	if d.Order == nil {
+		return 0, errors.New("don't know if big or little endian")
+	}
+	var v [4]byte
+	err := d.ReadRegRaw(reg, v[:])
+	return d.Order.Uint32(v[:]), err
+}
+
+// ReadRegUint64 reads a 64 bit register.
+func (d *DevReg) ReadRegUint64(reg byte) (uint64, error) {
+	if d.Order == nil {
+		return 0, errors.New("don't know if big or little endian")
+	}
+	var v [8]byte
+	err := d.ReadRegRaw(reg, v[:])
+	return d.Order.Uint64(v[:]), err
 }
 
 // WriteRegUint8 writes a 8 bit register.
-func (d *Dev) WriteRegUint8(reg byte, v uint8) error {
-	_, err := d.Write([]byte{reg, v})
-	return err
+func (d *DevReg) WriteRegUint8(reg byte, v uint8) error {
+	return d.Tx([]byte{reg, v}, nil)
 }
 
 // WriteRegUint16 writes a 16 bit register.
-func (d *Dev) WriteRegUint16(reg byte, v uint16) error {
-	_, err := d.Write([]byte{reg, byte(v >> 8), byte(v)})
-	return err
+func (d *DevReg) WriteRegUint16(reg byte, v uint16) error {
+	if d.Order == nil {
+		return errors.New("don't know if big or little endian")
+	}
+	var a [3]byte
+	a[0] = reg
+	d.Order.PutUint16(a[1:], v)
+	return d.Tx(a[:], nil)
+}
+
+// WriteRegUint32 writes a 32 bit register.
+func (d *DevReg) WriteRegUint32(reg byte, v uint32) error {
+	if d.Order == nil {
+		return errors.New("don't know if big or little endian")
+	}
+	var a [5]byte
+	a[0] = reg
+	d.Order.PutUint32(a[1:], v)
+	return d.Tx(a[:], nil)
+}
+
+// WriteRegUint64 writes a 64 bit register.
+func (d *DevReg) WriteRegUint64(reg byte, v uint64) error {
+	if d.Order == nil {
+		return errors.New("don't know if big or little endian")
+	}
+	var a [9]byte
+	a[0] = reg
+	d.Order.PutUint64(a[1:], v)
+	return d.Tx(a[:], nil)
 }
```

Comment:
WriteRegStruct could be implemented and could be useful to init a bunch of regs. But perhaps not worth it right now...